core/Lucy/Object/BitVector.c

```diff
@@ -46,6 +46,13 @@ static const uint32_t BYTE_COUNTS[256] = {
     4, 5, 5, 6, 5, 6, 6, 7, 5, 6, 6, 7, 6, 7, 7, 8
 };
 
+static CFISH_INLINE size_t
+SI_octet_size(size_t bit_size) {
+    if (bit_size > SIZE_MAX - 8) {
+        return SIZE_MAX / 8;
+    }
+    return (bit_size + 7) / 8;
+}
```

Comment:
The overflow check should be:

    if (bit_size > SIZE_MAX - 7) {
        return SIZE_MAX / 8 + 1;
    }

Or maybe an alternative implementation:

    static CFISH_INLINE size_t
    SI_octet_size(size_t bit_size) {
        if (bit_size == 0) { return 0; }
        return (bit_size - 1) / 8 + 1;
    }